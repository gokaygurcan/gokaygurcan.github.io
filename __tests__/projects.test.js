// gokaygurcan.github.io
//
// https://www.gokaygurcan.com/
// MIT © Gökay Gürcan

// node modules
import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

// local modules
import Projects from '../pages/projects';
import DummyProject from '../pages/projects/dummy';

describe('Enzyme', () => {
  it('Print "Projects page"', () => {
    const projects = shallow(
      <Projects />
    );

    expect(projects.find('span').text().trim()).toEqual('Projects page');
  });

  it('Print "Dummy Project"', () => {
    const dummyProjects = shallow(
      <DummyProject />
    );

    expect(dummyProjects.find('span').text().trim()).toEqual('Dummy Project');
  });
});

describe('Snapshot Testing', () => {
  it('Print "Projects page"', () => {
    const component = renderer.create(<Projects />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Print "Dummy Project"', () => {
    const component = renderer.create(<DummyProject />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
