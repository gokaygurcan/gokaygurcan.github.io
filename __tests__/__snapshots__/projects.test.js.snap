// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Print "Dummy Project" 1`] = `
<div
  className="wrapper"
  data-jsx={1407449589}
>
  <header
    data-jsx={1407449589}
  >
    <nav
      data-jsx={1407449589}
    >
      <a
        data-jsx={1407449589}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
        --  
      <a
        data-jsx={1407449589}
        href="/projects"
        onClick={[Function]}
      >
        Projects
      </a>
    </nav>
  </header>
  <span>
    Dummy Project
  </span>
  <footer
    data-jsx={1407449589}
  >
    MIT © Gökay Gürcan
  </footer>
</div>
`;

exports[`Snapshot Testing Print "Projects page" 1`] = `
<div
  className="wrapper"
  data-jsx={1407449589}
>
  <header
    data-jsx={1407449589}
  >
    <nav
      data-jsx={1407449589}
    >
      <a
        data-jsx={1407449589}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
        --  
      <a
        data-jsx={1407449589}
        href="/projects"
        onClick={[Function]}
      >
        Projects
      </a>
    </nav>
  </header>
  <div
    data-jsx={661177286}
  >
    <span
      data-jsx={661177286}
    >
      Projects page
    </span>
    <ul
      data-jsx={661177286}
    >
      <li
        data-jsx={661177286}
      >
        <a
          data-jsx={661177286}
          href="/projects/dummy"
          onClick={[Function]}
        >
          dummy
        </a>
      </li>
    </ul>
  </div>
  <footer
    data-jsx={1407449589}
  >
    MIT © Gökay Gürcan
  </footer>
</div>
`;
